runtime/syntax/sed.vim

```diff
@@ -19,6 +19,10 @@ syn region sedAddress   matchgroup=Special start="[{,;]\s*/\(\\/\)\="lc=1 skip="
 syn region sedAddress   matchgroup=Special start="^\s*/\(\\/\)\=" skip="[^\\]\(\\\\\)*\\/" end="/I\=" contains=sedTab,sedRegexpMeta
 syn match sedComment	"^\s*#.*$"
 syn match sedFunction	"[dDgGhHlnNpPqQx=]\s*\($\|;\)" contains=sedSemicolon,sedWhitespace
+if !exists("g:sed_syntax_style") || g:sed_syntax_style !=? "bsd"
+    syn match sedComment	"\s*#.*$"
+    syn match sedFunction	"[dDgGhHlnNpPqQx=]\s*" nextgroup=sedComment contains=sedSemicolon,sedWhitespace
```

Comment:
This is relaxing the definition of sedFunction so it now matches `d` in `djunk` which would otherwise be highlighted as an error. Maybe use something like the following so it only matches with tail comments.

```vim
syn match sedFunction "[dDgGhHlnNpPqQx=]\s*\ze#" nextgroup=sedComment contains=sedSemicolon,sedWhitespace
```

Reply:
added `\ze#` per your suggestion, see updated file